includes/classes/shopping_cart.php

```diff
@@ -786,7 +786,8 @@ function calculate() {
                 }
               if ($attribute_price->fields['attributes_discounted'] == '1') {
                 // calculate proper discount for attributes
-                $new_attributes_price = zen_get_discount_calc($product->fields['products_id'], $attribute_price->fields['products_attributes_id'], $attribute_price->fields['options_values_price'], $qty);
+                $new_attributes_price = zen_get_discount_calc($product->fields['products_id'], $attribute_price->fields['products_attributes_id'], /*$attribute_price->fields['options_values_price']*/ $attribute_price->fields["options_values_price"] + (zen_get_products_price_is_priced_by_attributes($attribute_price->fields["products_id"]) ? zen_products_lookup($attribute_price->fields["products_id"], 'products_price') : 0), $qty);
```

Comment:
I believe at the moment that this upgrade affecting commenting is more about the development changes rather than the final product by which others would use to finally get the expected prices displayed/processed when working with attributes (minus perhaps unique tax classes on an attribute). I know that for me it has provided a sort of marker that the code has been touched in that area and what it originally was. For such inline replacement I would think the redundant commented out code should be removed for final producr. 

With the breadth of impact overall (additional processing when displaying values, additional queries, etc.) that some amount of supporting review/operations need to be confirmed. I've tried to methodically go through variations, but I wouldn't doubt that I have missed some expectancy.